lib/inets/src/http_client/httpc_handler.erl

```diff
@@ -622,12 +622,12 @@ do_handle_info({ssl_closed, _}, State = #state{request = undefined}) ->
     {stop, normal, State};
 
 %%% Error cases
-do_handle_info({tcp_closed, _}, #state{session = Session0} = State) ->
+do_handle_info({tcp_closed, _}, #state{session = Session0} = State) when Session0 =/= undefined->
     Socket  = Session0#session.socket,
     Session = Session0#session{socket = {remote_close, Socket}},
     %% {stop, session_remotly_closed, State};
     {stop, normal, State#state{session = Session}};
-do_handle_info({ssl_closed, _}, #state{session = Session0} = State) ->
+do_handle_info({ssl_closed, _}, #state{session = Session0} = State) when Session0 =/= undefined->
```

Comment:
Same as above!

Reply:
Well I guess this goes for the whole file. I think it is better to assert that you get the expected type by matching the expected record type `#session{}, #request{} ,  #http_response_h{},` than checking that it is not undefined.